cvat/apps/quality_control/quality_reports.py

```diff
@@ -2330,19 +2666,20 @@ def schedule_custom_quality_check_job(
                     queue, user_id=user_id, rq_id=rq_id, should_be_dependent=True
                 )
 
-                queue.enqueue(
-                    self._check_task_quality,
-                    task_id=task.id,
-                    job_id=rq_id,
-                    meta=BaseRQMeta.build(request=request, db_obj=task),
-                    result_ttl=self._JOB_RESULT_TTL,
-                    failure_ttl=self._JOB_RESULT_TTL,
-                    depends_on=dependency,
-                )
+            job_callback, job_params = self._make_rq_job_params(target=target)
+            queue.enqueue(
+                job_callback,
+                **job_params,
+                job_id=rq_id,
+                meta=BaseRQMeta.build(request=request, db_obj=target),
+                result_ttl=self._JOB_RESULT_TTL,
+                failure_ttl=self._JOB_RESULT_TTL,
+                depends_on=dependency,
+            )
```

Comment:
Why did you move job enqueuing outside a lock? It must be done inside a critical section. Otherwise, 2 parallel requests from the same user to compute quality reports for different resources may have the same dependency.

Reply:
It looks like an invalid automatic merge. Thanks for noticing